themes/mlibrary_new/custom.php

```diff
@@ -130,16 +145,42 @@ function mlibrary_new_display_exhibit_card_info($exhibitPage)
   ];
 }
 
-// get the type of the item
-function mlibrary_new_display_exhibit_type_of_item($rawAttachment)
+//gallery
+function mlibrary_new_get_image_for_gallery($attachment = null)
 {
-   $itemType = 'Still Image';
+   return record_image($attachment->getFile(),'original',array('class' => 'image-card'));
+}
 
-   if ($rawAttachment[0]->getItem()->getItemType()!=null) {
-       $itemType = $rawAttachment[0]->getItem()->getItemType()->name;
- }
-   
-   return $itemType;
+function mlibrary_new_create_card_for_gallery($attachment)
+{
+   return [
+     'image' => mlibrary_new_get_image_for_gallery($attachment),
+     'title' => get_view()->shortcodes(metadata($attachment,'caption',array('no_escape' => true))),
+   ];
+}
+
+function mlibrary_new_render_gallery_section($sectionpage_cards_info){
+  foreach ($sectionpage_cards_info as $sectionpage_card_info) {
+      $html = '<div id = "exhibit-theme-item" class="panel panel-default">';
+      $html .= '<div class="panel-heading">'.$sectionpage_card_info["image"].'</div>';
+      $html .= '<div class="card-info panel-body"><h3 class="panel-card-title">'.$sectionpage_card_info["title"].'</h3></div>';
+      $html .= '</div>';
+      echo $html;
```

Comment:
Lines 164-168 look problematic to me.

I advise against creating functions with side-effects, like writing output.  I recommend putting it all in a string and returning the string.

If you do go the route of directly printing the html from this function, then I advise against buffering it to a string.